samza-core/src/main/java/org/apache/samza/config/ClusterManagerConfig.java

```diff
@@ -87,7 +88,10 @@
    */
   public static final String CONTAINER_RETRY_COUNT = "yarn.container.retry.count";
   public static final String CLUSTER_MANAGER_CONTAINER_RETRY_COUNT = "cluster-manager.container.retry.count";
-  private static final int DEFAULT_CONTAINER_RETRY_COUNT = 8;
+  public static final int DEFAULT_CONTAINER_RETRY_COUNT = 8;
+
+  public static final String CLUSTER_MANAGER_CONTAINER_RETRY_MAX_DELAY_MS = "cluster-manager.container.host-affinity-retry.max.delay.ms";
+  public static final long DEFAULT_CONTAINER_RETRY_MAX_DELAY_MS = Duration.ofSeconds(120).toMillis();
```

Comment:
Debating if this should be a non-configurable constant. Thoughts?

Reply:
This should be in greater than the max amount of time it takes for the RM to mark a dead NM unhealthy, 
which in YARN 2.9.2 is 5 minutes?
So perhaps not configurable and set to a higher value.

Reply:
Yes maybe add 1 second to account for clock-skew and asynchrony. 
(typically clock skew with a dc is 100s of micros).

Reply:
Sure will add